test/aria/widgets/form/multiautocomplete/BaseMultiAutoCompleteTestCase.js

```diff
@@ -59,11 +63,24 @@ Aria.classDefinition({
 
         type : function (evt, args) {
             args = args || evt;
-            this.synEvent.type(this.getFocusedElement(), args.text.shift(), {
+
+            var nextText = args.text.shift();
+
+            var cb = {
                 fn : this.__wait,
                 scope : this,
                 args : args
-            });
+            };
+
+            if (typeof(nextText) == "function") {
+                this.waitFor({
+                    condition : nextText,
+                    callback : cb
+                });
+            } else {
+                // nextText is a string
+                this.synEvent.type(this.getFocusedElement(), nextText, cb);
+            }
```

Comment:
I've realized there might be some kind of confusion happening here:

The `clickAndType` method here overrides `clickAndType` from TemplateTestCase [2] (since this class extends RobotTestCase which extends TemplateTestCase)

Hence adding the feature of accepting callback parameters will only work for MAC tests, not all Template tests. It would probably be interesting to move this feature to the TTC class and refactor the code of  `clickAndType` here and in the parent class accordingly.

[1] https://github.com/divdavem/ariatemplates/blob/randomlyFailingTests/test/aria/widgets/form/multiautocomplete/BaseMultiAutoCompleteTestCase.js#L45-84
[2] https://github.com/divdavem/ariatemplates/blob/randomlyFailingTests/src/aria/jsunit/TemplateTestCase.js#L702-725